### DISCUSSION TITLE
feel the other language lack with the support of competitive programming data structures

### DISCUSSION THREAD

**Discussion by vincenzopalazzo** [vlang/v#14625] [Ideas]
I think that the new languages has a leak in the implementation of the std, that is the possibility to have complex data structure implemented directly in the language.

This can help to feel a lack that is in all the language.

I built some things in https://github.com/vincenzopalazzo/cpstl for my purpose, and I'm thinking that maybe we can build some of these data structures there?

**Comment by playXE** [comment]
There is already vlib/datatypes module and cpstl can be ported into it IMO

**Comment by vincenzopalazzo** [comment]
>There is already vlib/datatypes module and cpstl can be ported into it IMO

This is exactly my point
